mcp/sensors/star_camera_receive.c

```diff
@@ -0,0 +1,347 @@
+    } else if (which == 2 && CommandData.sc_resets.reset_sc2_param) {
+        CommandData.sc_resets.reset_sc2_param = 0;
+        return 1;
+    } else {
+        return 0;
+    }
+}
+
+// Now that we have the general boilerplate and useful functions out of the way
+// I will add the thread function that
+
+void *parameter_receive_thread(void *args) {
+    struct socket_data * socket_target = args;
+    int first_time = 1;
+    int sockfd;
+    struct addrinfo hints, *servinfo, *servinfoCheck;
+    int returnval;
+    int numbytes;
+    int listening = 1;
+    int which_sc;
+    struct sockaddr_storage their_addr;
+    struct star_cam_return received_parameters;
+    socklen_t addr_len;
+    char ipAddr[INET_ADDRSTRLEN];
+    memset(&hints, 0, sizeof hints);
+    hints.ai_family = AF_INET; // set to AF_INET to use IPv4
+    hints.ai_socktype = SOCK_DGRAM;
+    hints.ai_flags = AI_PASSIVE; // use my IP
+    int err;
+    if (!strcmp(socket_target->ipAddr, SC1_IP_ADDR)) {
+        which_sc = 1;
+        blast_info("Param socket pointed at %s:%s\n", socket_target->ipAddr, socket_target->port);
+    } else if (!strcmp(socket_target->ipAddr, SC2_IP_ADDR)) {
+        which_sc = 2;
+        blast_info("Param socket pointed at %s:%s\n", socket_target->ipAddr, socket_target->port);
+    } else {
+        blast_info("Invalid IP target for data source\n");
+        return NULL;
+    }
+    // Fix all this here
+    while (listening) {
+        listening = check_sc_thread_bool(which_sc);
+        if (first_time) {
+            first_time = 0;
+            if ((returnval = getaddrinfo(NULL, socket_target->port, &hints, &servinfo)) != 0) {
+                blast_err("getaddrinfo: %s\n", gai_strerror(returnval));
+                return NULL;
+            }
+            // loop through all the results and bind to the first we can
+            for (servinfoCheck = servinfo; servinfoCheck != NULL; servinfoCheck = servinfoCheck->ai_next) {
+                // since a success needs both of these, but it is inefficient to do fails twice,
+                // continue skips the second if on a fail
+                if ((sockfd = socket(servinfoCheck->ai_family, servinfoCheck->ai_socktype,
+                        servinfoCheck->ai_protocol)) == -1) {
+                    perror("listener: socket");
+                    continue;
+                }
+                if (bind(sockfd, servinfoCheck->ai_addr, servinfoCheck->ai_addrlen) == -1) {
+                    close(sockfd);
+                    perror("listener: bind");
+                    continue;
+                }
+                break;
+            }
+            if (servinfoCheck == NULL) {
+                blast_err("listener: failed to bind socket\n");
+                return NULL;
+            }
+            // set the read timeout (if there isnt a message) to 100ms
+            struct timeval read_timeout;
+            read_timeout.tv_sec = 0;
+            read_timeout.tv_usec = 100000;
```

Comment:
No magic numbers! Define a function-level or header-level variable called e.g. `read_timeout_usec` and use that instead of `100000`. I like that a comment explains this, but giving the timeout a name does the job of a comment and relieves us of the maintenance burden of having to make sure the code and comment match if the timeout is ever changed.

Reply:
done